public/app/features/plugins/admin/helpers.ts

```diff
@@ -469,3 +472,84 @@ export function isPluginUpdateable(plugin: CatalogPlugin) {
 
   return true;
 }
+
+export function getPluginRepositoryUrl(
```

Comment:
I might a bit late to it, and it might be a next iteration kind of thing:
I would move this and the branch fetching logic into grafana.com/api/plugins - something like

https://grafana.com/api/plugins/grafana-clock-panel/links and then just fetch those in grafana and handle the display logic.

This way it would be decoupled from the core, cacheable and less direct requests against github / gitlab coming from the users browser.

Reply:
+1 on this. This will make it much easier for us to change the logic on how we fetch these links